pycytominer/cyto_utils/cells.py

```diff
@@ -483,12 +525,73 @@ def aggregate_compartment(
                     )
                 )
 
+                if add_image_features:
```

Comment:
This feels very complicated!

If it DOES in fact need to be this complicated, I think it might be time to split out this functionality into a new function...

Let's discuss this here and if we decide it does need this complexity, then i'll review in more detail after

Reply:
It does look very complicated. Let me split this section into a couple of new functions and get back to you.

Reply:
I have done so in https://github.com/cytomining/pycytominer/pull/168/commits/6f247cd394e3e7fbbb4ce41d4dd68b88243e2730. Let me know what you think.